tests/unit/obuf-limits.tcl

```diff
@@ -70,4 +70,32 @@ start_server {tags {"obuf-limits"}} {
         assert {$omem >= 100000 && $time_elapsed < 6}
         $rd1 close
     }
+
+    test {Won't write replies if client output buffer hard limit is enforced} {
+        r config set client-output-buffer-limit {normal 100000 0 0}
+
+        # Total size of all items must be more than 2 * 100000
+        for {set i 0} {$i < 100000} {incr i} {
+            r lpush mylist $i $i
+        }
+        set orig_mem [s used_memory]
+
+        # Get all items
+        set rd [redis_deferring_client]
+        $rd lrange mylist 0 -1
+
+        # Redis will not use 2 * 100000 memory to reply because we won't
+        # write any things if reached client output buffer hard limit.
+        set current_mem [s used_memory]
+        assert {$current_mem < 2 * 100000 + $orig_mem}
+
+        # We won't read anything about mylist
+        catch { [$rd read] } e
+        assert_no_match "*mylist*" $e
+        assert_match "*I/O error*" $e
+
+        # Client memory is released
+        set released_mem [s used_memory]
+        assert {$released_mem < $orig_mem + 100000}
```

Comment:
unless i'm missing something, i guess this test isn't needed (since it'll be covered already, before we did `$rd read`)

Reply:
Yes, i need to remove it

Reply:
Yes, i remove it